### PR TITLE
Consolidate item-release robots

### PR DESCRIPTION
**Pull request by jcoyne**
Note: Need to ensure shared_configs are consistent before deploying, see #172

### CODE REVIEW

File: lib/dor/item.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      create_workflow(druid)
+    end
+
+    def self.create_workflow(druid)
+      LyberCore::Log.debug "...adding workflow #{Dor::Config.release.workflow_name} for #{druid}"
+
+      # initiate workflow
+      with_retries(max_tries: Dor::Config.release.max_tries, base_sleep_seconds: Dor::Config.release.base_sleep_seconds, max_sleep_seconds: Dor::Config.release.max_sleep_seconds) do |_attempt|
+        obj = Dor.find(druid)
+        obj.create_workflow(Dor::Config.release.workflow_name)
```
Comment (jcoyne) [inline lib/dor/item.rb]: this is out of date, use https://github.com/sul-dlss/item-release/blob/d75ce01d082f58f48363dbdad406676a61f9773a/lib/dor/item.rb#L102

File: config/deploy.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@
 ask :branch, proc { `git rev-parse --abbrev-ref HEAD`.chomp }
 
 # Default deploy_to directory is /var/www/my_app
-set :deploy_to, "/home/lyberadmin/#{fetch(:application)}"
+set :deploy_to, "/opt/app/lyberadmin/#{fetch(:application)}"
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply config/deploy.rb]: yes

File: config/deploy/stage.rb [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-server 'sul-robots2-test.stanford.edu', user: 'lyberadmin', roles: %w{web app}
-
+server 'common-accessioning-stage-a.stanford.edu', user: 'lyberadmin', roles: %w{web app db} # only server to get whenever cronjobs deployed to
+server 'common-accessioning-stage-b.stanford.edu', user: 'lyberadmin', roles: %w{web app}
 
 Capistrano::OneTimeKey.generate_one_time_key!
 
-set :deploy_environment, 'test'
+# This gets set as the ROBOT_ENVIRONMENT veriable in crontab (via whenever)
+set :deploy_environment, 'production'
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply config/deploy/stage.rb]: Ah, good call.

[the remark this answers is not in the document]
Comment (jcoyne) [reply config/deploy/stage.rb]: Oh, this is supposed to be production.

File: config/boot.rb [hunk 4]
```diff
@@ -33,6 +33,11 @@
 puts "Loading config from #{env_file}"
 require env_file
 
+# Configure dor-services-client to use the dor-services URL
```
Comment (ndushay) [inline config/boot.rb]: do we need the comment?  the code reads cleanly

File: config/deploy.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (ndushay) [reply config/deploy.rb]: is there a plan to not be in lyberadmin?  (I'm fine either way)

File: config/deploy/prod.rb [hunk 5]
```diff
@@ -1,10 +1,10 @@
 # frozen_string_literal: true
 
-server 'sul-robots1-prod.stanford.edu', user: 'lyberadmin', roles: %w{web app db}
-server 'sul-robots2-prod.stanford.edu', user: 'lyberadmin', roles: %w{web app}
-server 'sul-robots3-prod.stanford.edu', user: 'lyberadmin', roles: %w{web app}
-server 'sul-robots4-prod.stanford.edu', user: 'lyberadmin', roles: %w{web app}
-server 'sul-robots5-prod.stanford.edu', user: 'lyberadmin', roles: %w{web app}
+server 'common-accessioning-prod-a.stanford.edu', user: 'lyberadmin', roles: %w{web app db} # only server to get whenever cronjobs deployed to
```
Comment (ndushay) [inline config/deploy/prod.rb]: should there be a cap "whenever" role to make this explicit?

File: config/deploy/stage.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (ndushay) [reply config/deploy/stage.rb]: Yeah - the "stage" machines are in rails "production"

File: lib/dor/item.rb [hunk 6]
```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require 'retries'
+require 'dor-fetcher'
+
+module Dor::Release
```
Comment (ndushay) [inline lib/dor/item.rb]: should this live in `lib/dor/release/item` ?

File: robots/release/release_members.rb [hunk 7]
```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# Robot class to run under multiplexing infrastructure
+module Robots
+  module DorRepo
```
Comment (ndushay) [inline robots/release/release_members.rb]: similar question about directory structure adhering to implied containment in code.  Here, important only if preservation robots come in.  (TBH, I hate the "sdr" vs "dor" in the robots and would be happy for it to go away)

File: config/deploy/prod.rb (hunk 5 above)
Comment (jcoyne) [reply config/deploy/prod.rb]: Whenever has to happen on all of the servers, but one whenever thing only happens on one server.  This hasn't changed from how we currently have it.

File: config/boot.rb (hunk 4 above)
Comment (jcoyne) [reply config/boot.rb]: Peter thought this was helpful. 🤷‍♂️  https://github.com/sul-dlss/common-accessioning/commit/c474925fb119643c3df828bc17fa4ff1578d49c4#diff-dfab7ed9f3a01a0f58ab1de265a6ae91R36

File: lib/dor/item.rb (hunk 6 above)
Comment (jcoyne) [reply lib/dor/item.rb]: Currently it's https://github.com/sul-dlss/item-release/blob/master/lib/dor/item.rb.  The idea is to not make too many changes here.

File: robots/release/release_members.rb (hunk 7 above)
Comment (jcoyne) [reply robots/release/release_members.rb]: This is just a straight copy from https://github.com/sul-dlss/item-release/blob/master/robots/release/release_members.rb

File: config/deploy.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcoyne) [reply config/deploy.rb]: This is the directory on the new machines:
```
$ echo $HOME
/opt/app/lyberadmin
```

**Pull request merged**